parts/linux/cloud-init/nodecustomdata.yml

```diff
@@ -350,6 +350,22 @@ write_files:
     net.bridge.bridge-nf-call-iptables = 1
     #EOF
 
+{{- if IsKubenet }}
+- path: /usr/local/bin/configure_masq_rules.sh
+  permissions: "0755"
+  owner: root
+  content: |
+    #!/bin/bash
+    set -x
+    # we are currently enabling ipmasq for kubenet cni with containerd which has the unfortunate
+    # side effect of SNATing all outgoing pod-to-pod traffic to node ip
+    # kubelet does not handle non-masquerade-cidr for containerd so we will have to handle it ourselves here
+    # this iptable rule allows all traffic heading to the cluster vnets to avoid SNAT
+    # TODO remove this after we transition kubenet clusters to use ip-masq-agents
+    iptables -t nat -I POSTROUTING 1 -m addrtype ! --dst-type LOCAL -d {{GetParameter "vnetCidr"}} -j ACCEPT
```

Comment:
do we need ! --dst-range 168.63.129.16 in here? I think no since it's except not masquerade like line 483.

Reply:
Also the name vnetCidr is very confusing. I believe its actually the pod cidr? In azure cni's case that is also the vnet cidr (kinda) but in kubenet it's differnt. Have you confirmed it's the pod cidr in kubenet cases?

Reply:
yea I used the wrong var here. Updated to use `kubeClusterCidr` which uses the ClusterSubnet field in our model. That field is mapped to/from cluster podCidr for kubenet and also what we pass into --non-masq-cidr for kubelet

Reply:
re: 168.63.129.16 - I think no as well? That line you ref'ed is only for azureCNI...I suppose since all pod/node IPs are on the same subnet for kubenet we don't need to masq source IPs to talk to  168.63.129.16

Also b/c we haven't done it historically for docker so I don't see why we'd need to do it for containerd